### PR TITLE
Main compose packagesets

### PR DESCRIPTION
**Pull request by bcl**
This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change
- [ ] PR contains automated tests for new functionality and

### CODE REVIEW

**Comment by ondrejbudai** [comment]
This PR modified the signature of `Store.PushCompose()` but did not modify the calls, therefore the tests are failing.

I think we also need to add `PackageSets` to `internal/store/json.go` so the package sets are persistent.

File: internal/weldr/api.go [hunk 1]
```diff
@@ -2370,6 +2365,18 @@ func (api *API) composeInfoHandler(writer http.ResponseWriter, request *http.Req
 		reply.Uploads = targetsToUploadResponses(compose.ImageBuild.Targets, composeStatus.State)
 	}
 
+	// Add package dependencies from the compose
+	// This information may not be included for the compose
+	if _, ok := compose.PackageSets["packages"]; ok {
+		dependencies := compose.PackageSets["packages"]
```
Comment (teg) [inline internal/weldr/api.go:2371]: Could we just store the `packages` PackageSet in the store? That way we don't make the names of the sets API, and we would also avoid storing the build-root packages etc which we currently have no use for / way to export.

File: cmd/osbuild-store-dump/main.go [hunk 2]
```diff
@@ -170,11 +170,11 @@ func main() {
 	if err != nil {
 		panic(err)
 	}
-	manifest, packageSpecSets = getManifest(bp2, t2, a, d, rpmmd, repos)
+	manifest, packageSpecSets = getManifest(bp1, t2, a, d, rpmmd, repos)
```
Comment (teg) [inline cmd/osbuild-store-dump/main.go:173]: Could you explain the context a bit in the commit message? :)

File: internal/store/json.go [hunk 3]
```diff
@@ -29,8 +30,9 @@ type workspaceV0 map[string]blueprint.Blueprint
 // It contains all the information necessary to generate the inputs for the job, as
 // well as the job's state.
 type composeV0 struct {
-	Blueprint   *blueprint.Blueprint `json:"blueprint"`
-	ImageBuilds []imageBuildV0       `json:"image_builds"`
+	Blueprint   *blueprint.Blueprint           `json:"blueprint"`
+	ImageBuilds []imageBuildV0                 `json:"image_builds"`
+	PackageSets map[string][]rpmmd.PackageSpec `json:"package_sets"`
```
Comment (teg) [inline internal/store/json.go:35]: As suggested elsewhere: should we make this only be the `packages` set, so we don't make the names of the sets API, nor the build-root++

**Review by teg — CHANGES_REQUESTED** [review]
I really like this! Just one suggestion for changes inline.

**Review by teg — APPROVED** [review]
Nice! Thanks!

**Review by ondrejbudai — APPROVED** [review]

**Comment by bcl** [comment]
@teg Can this be merged? I don't see why Schutzbot says failed, everything looks green on the details page.

**Comment by teg** [comment]
Base tests timed out on aarch64 EL8. This is caused by `qemu-img` hanging, not related to this PR.

**Pull request merged**